src/ColinHDev/CPlot/provider/DataProvider.php

```diff
@@ -1299,4 +1313,213 @@ public function savePlotRate(Plot $plot, PlotRate $plotRate) : Generator {
+                yield from DataProvider::getInstance()->savePlotPlayer($plot, $senderData);
+
+                // load helpers
+                foreach($record["helpers"] as $playerName) {
+                    // validate offline player data
+                    $offlineData = Server::getInstance()->getOfflinePlayerData($playerName);
+                    $XUID = $offlineData?->getString("LastKnownXUID", null);
+
+                    // register filler player data
+                    /** @var PlayerData|null $playerData */
+                    $playerData = yield $this->updatePlayerData(
+                        null, // doesn't matter what is input at this point. will overwrite on login
+                        $XUID,
+                        $playerName
+                    );
+                    if (!($playerData instanceof PlayerData)) {
+                        // retry now that player data updated
+                        $playerData = yield $this->awaitPlayerDataByData(
+                            null,
+                            $XUID,
+                            $playerName
+                        );
+                    }
+
+                    $senderData = new PlotPlayer($playerData, PlotPlayer::STATE_TRUSTED);
+                    $plot->addPlotPlayer($senderData);
+                    yield from DataProvider::getInstance()->savePlotPlayer($plot, $senderData);
+                }
+
+                // load denied with priority over helpers
+                foreach($record["denied"] as $playerName) {
+                    // validate offline player data
+                    $offlineData = Server::getInstance()->getOfflinePlayerData($playerName);
+                    $XUID = $offlineData?->getString("LastKnownXUID", null);
+
+                    // register filler player data
+                    /** @var PlayerData|null $playerData */
+                    $playerData = yield $this->updatePlayerData(
+                        null, // doesn't matter what is input at this point. will overwrite on login
+                        $XUID,
+                        $playerName
+                    );
+                    if (!($playerData instanceof PlayerData)) {
+                        // retry now that player data updated
+                        $playerData = yield $this->awaitPlayerDataByData(
+                            null,
+                            $XUID,
+                            $playerName
+                        );
+                    }
+
+                    $senderData = new PlotPlayer($playerData, PlotPlayer::STATE_DENIED);
+                    $plot->addPlotPlayer($senderData);
+                    yield from DataProvider::getInstance()->savePlotPlayer($plot, $senderData);
+                }
+
+                //load common flags
+                $flag = Flags::PVP()->createInstance($record["pvp"]);
+                $flag = $plot->getLocalFlagByID($flag->getID())?->merge($flag->getValue()) ?? $flag;
```

Comment:
The `getLocalFlagByID()` method returns a flag instance if it was set by a player. Since this plot should just be created, I don't think there is a reason to check for existing flags